### PR TITLE
DM-13618: Find node-hour usage from usage codes

### PR DESCRIPTION
**Pull request by Samantha-Thrush**
I've added a calculation for node-hours in the usageploy.py code in order easily see how many resources are spent on a given set of slurm jobs.   In order to include this calculation, I've added in a function to calculate node-hours, and have split the previous plotting function definition into two function definition: one for extracting data from the data_file into numpy arrays, and one for creating the plot.

### CODE REVIEW

File: usage/usageplot.py [hunk 1]
```diff
@@ -66,45 +68,94 @@ def get_args(args):
 
     return title, datafile, name
```
Comment (mxk62) [inline usage/usageplot.py:74]: All functions (and classes) must be separated by *two* blank lines from each other and any other parts of the module (e.g. imports).

File: usage/usageplot.py [hunk 2]
```diff
@@ -1,4 +1,6 @@
 """Create a plot of the lsst-dev node useage over time
+Prints the total node_hours of the jobs provided to
+usage.py.
```
Comment (mxk62) [inline usage/usageplot.py]: Class/function/module summary should preferably be a short, one-line sentence. So either you're forgot to put a blank line between the first and the second line or it is not a good summary. ;)

File: usage/usageplot.py [hunk 3]
```diff
@@ -66,45 +68,94 @@ def get_args(args):
 
     return title, datafile, name
 
-def make_plot(title, data_file, name):
-    """ Gets the title and data_file name and makes a time vs. NNode plot named usage.png
+def make_plot(title, times, nodes, name):
+    """ Gets the title, the time and node arrays and 
+    makes a time vs. NNode plot named 'name'.png.
```
Comment (mxk62) [inline usage/usageplot.py]: See my earlier remark regarding summary lines in docstrings.

Also there should be *no* space space between opening """ and the first word of the summary. Please fix it in other docstrings too.

File: usage/usageplot.py [hunk 4]
```diff
@@ -66,45 +68,94 @@ def get_args(args):
...
+    """
+    prev_time = 0
+    node_hours = 0
+    for i in range(len(times)):
+        dur = times[i] - prev_time
+        prev_time = times[i]
+        x = nodes[i]*dur
+        node_hours += x
+    return node_hours
+
```
Comment (mxk62) [inline usage/usageplot.py]: Time intervals used in `usage.py` are all equal. This greatly simplifies calculation of the metric you are interested in:
```python
def get_nodehours(times, nodes):
    """...
    """
    dt = times[1] - times[0]
    return np.sum(dt * nodes)
```

**Review by mxk62 — CHANGES_REQUESTED** [review]
Firstly, you're making a cardinal sin (;D) mixing spaces and tabs in Python code. Please, configure editor you are using to expand each tabs into spaces (each tab into 4).

Secondly, I cannot see any consistency with regard to the length of the lines in your docstrings. Some of sentences/paragraphs span to the "legal" limit (79 characters) as they should, others seems to be broken half way through. Please, fix it.

Finally, and I apologize for not catching it earlier, the standard types in docstrting (e.g. `str`, `float`) should be enclosed in ` (backticks), not in ' (single quotes), see [this]( https://developer.lsst.io/docs/py_docs.html#describing-parameter-types) part of the official DM Dev Guide.

**Comment by Samantha-Thrush** [comment]
Thank you so much for the review, Mikolaj!  I have resolved all of the issues above and so I think it's ready for your final review!

**Review by mxk62 — APPROVED** [review]
You can merge it to master providing you fix issues I listed below:

1. There are still some leftover tabs (lines: 31, 154), remove them.
2. Insert a blank line after line 17.
3. Move `matplotlib.use('Agg')` after the imports (remember about separating two blank lines).
4. Put a space after `__name__` in line 152.

**Pull request merged**